packages/dev/loaders/src/USD/usdFileLoader.ts

```diff
@@ -0,0 +1,198 @@
+import type { ISceneLoaderPluginAsync, ISceneLoaderPluginFactory, ISceneLoaderAsyncResult, ISceneLoaderProgressEvent } from "core/Loading/sceneLoader";
+import { registerSceneLoaderPlugin } from "core/Loading/sceneLoader";
+import { USDFileLoaderMetadata } from "./usdFileLoader.metadata";
+import { AssetContainer } from "core/assetContainer";
+import type { Scene } from "core/scene";
+import type { Nullable } from "core/types";
+import type { AbstractMesh } from "core/Meshes/abstractMesh";
+import { Mesh } from "core/Meshes/mesh";
+import { VertexData } from "core/Meshes/mesh.vertexData";
+import { RawTexture } from "core/Materials/Textures/rawTexture";
+import { StandardMaterial } from "core/Materials/standardMaterial";
+import { Texture } from "core/Materials/Textures/texture";
+import { Engine } from "core/Engines/engine";
+import { Constants } from "core/Engines/constants";
+import type { USDLoadingOptions } from "./usdLoadingOptions";
+import { _LoadScriptModuleAsync } from "../shared/tools.internal";
+
+declare module "core/Loading/sceneLoader" {
+    // eslint-disable-next-line jsdoc/require-jsdoc
+    export interface SceneLoaderPluginOptions {
+        /**
+         * Defines options for the usd loader.
+         */
+        [USDFileLoaderMetadata.name]: Partial<USDLoadingOptions>;
+    }
+}
+
+/**
+ * @experimental
+ * USD(z) file type loader.
+ * This is a babylon scene loader plugin.
+ */
+export class USDFileLoader implements ISceneLoaderPluginAsync, ISceneLoaderPluginFactory {

```

Comment:
For new code, I would consider not having the plugin itself implement `ISceneLoaderPluginFactory`. You'll want to add a factory to `packages\dev\loaders\src\dynamic.ts`, so really you could follow the same pattern for the `registerSceneLoaderPlugin` side effect.